distutils/dist.py

```diff
@@ -1125,20 +1124,32 @@ def write_pkg_file(self, file):
                 self.classifiers or self.download_url):
             version = '1.1'
 
+        # required fields
         file.write('Metadata-Version: %s\n' % version)
         file.write('Name: %s\n' % self.get_name())
         file.write('Version: %s\n' % self.get_version())
-        file.write('Summary: %s\n' % self.get_description())
-        file.write('Home-page: %s\n' % self.get_url())
-        file.write('Author: %s\n' % self.get_contact())
-        file.write('Author-email: %s\n' % self.get_contact_email())
-        file.write('License: %s\n' % self.get_license())
+
+        # optional fields
+        summary = self.get_description()
+        if summary:
+            file.write('Summary: %s\n' % summary)
+        home_page = self.get_url()
+        if home_page:
+            file.write('Home-page: %s\n' % home_page)
+        author = self.get_contact()
+        if author:
+            file.write('Author: %s\n' % author)
+        author_email = self.get_contact_email()
+        if author_email:
+            file.write('Author-email: %s\n' % author_email)
+        license = self.get_license()
+        if license:
+            file.write('License: %s\n' % license)
         if self.download_url:
             file.write('Download-URL: %s\n' % self.download_url)
-
-        long_desc = rfc822_escape(self.get_long_description())
```

Comment:
I have the impression that RFC 822 escaping is required by the core metadata spec: https://packaging.python.org/en/latest/specifications/core-metadata/#description

Maybe the escaping in distutils differs from the one in the spec, instead of 8 spaces, it should be 7 spaces and a |, but no escaping might confuse the libraries reading the file.